src/termination.rs

```diff
@@ -1,9 +1,10 @@
 use crate::*;
 
-#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize)]
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Clone)]
 #[serde(crate = "near_sdk::serde")]
-#[cfg_attr(not(target_arch = "wasm32"), derive(Debug, PartialEq, Clone))]
+#[cfg_attr(not(target_arch = "wasm32"), derive(Debug, PartialEq))]
 pub enum HashOrSchedule {
+    SameAsLockupSchedule,
```

Comment:
@evgenykuzyakov, honestly, I see no point keeping the backward compatibility right now.
There is no self upgrade functionality inside the current contract, so the only option is to upgrade it via full access key.
In that case, any code can be deployed, any transaction can be signed on behalf of the lockup contract. It means the lockup contract does not serve it's purpose (it does not guarantee that locked tokens will be available for you to claim).

@frol , does anyone use this lockup code in production?
Do we have any plans to allow the upgrades?
Do we need to keep backward compatibility at this stage?

I thought that instead of upgrades we would rather create a new lockup contract with new code, totally unrelated to the original one.

Reply:
As we discussed with @evgenykuzyakov, there is one known production use of this contract, but I agree that without proper upgradability we can ignore the breaking API concern. @evgenykuzyakov suggested offline that we bump the version to indicate the breaking change and call it a day, and I agree with him.